### ISSUE TITLE
saved-searches example program exception when calling AtomEntry.cs

### ISSUE THREAD

**Issue by machdevnet** [splunk/splunk-sdk-csharp-pcl#51]
When running the C# SDK release 2.2.2 against Splunk Server v6.3.3, the saved-searches example program fails and throws an exception when calling AtomEntry in Splunk.Client. See the exception details below. The cause appears to be an attempted duplicate **_`propertyName`_** in **AtomEntry** and its subsequent addition in the **_`dictionary`_**, at or around **line 396**.

```
System.AggregateException was unhandled
  HResult=-2146233088
  Message=One or more errors occurred.
  Source=mscorlib
  StackTrace:
       at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
       at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
       at System.Threading.Tasks.Task.Wait()
       at Splunk.Examples.saved_searches.Program.Main(String[] args) in C:\Users\jdhea\Documents\Visual Studio 2015\Projects\splunk-sdk-csharp-pcl-2.2.2\examples\saved-searches\Program.cs:line 33
       at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
       at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
       at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
       at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
       at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
       at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
       at System.Threading.ThreadHelper.ThreadStart()
  InnerException: 
       HResult=-2146233087
       Message=Found invalid data while decoding.
       Source=Splunk.Client
       StackTrace:
            at Splunk.Client.AtomEntry.<ParseDictionaryAsync>d__32.MoveNext() in C:\Users\jdhea\Documents\Visual Studio 2015\Projects\splunk-sdk-csharp-pcl-2.2.2\src\Splunk.Client\Splunk\Client\AtomEntry.cs:line 396
         --- End of stack trace from previous location where exception was thrown ---
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
            at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
            at Splunk.Client.AtomEntry.<ParsePropertyValueAsync>d__34.MoveNext() in C:\Users\jdhea\Documents\Visual Studio 2015\Projects\splunk-sdk-csharp-pcl-2.2.2\src\Splunk.Client\Splunk\Client\AtomEntry.cs:line 468
         --- End of stack trace from previous location where exception was thrown ---
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
            at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
            at Splunk.Client.AtomEntry.<ReadXmlAsync>d__29.MoveNext() in C:\Users\jdhea\Documents\Visual Studio 2015\Projects\splunk-sdk-csharp-pcl-2.2.2\src\Splunk.Client\Splunk\Client\AtomEntry.cs:line 236
         --- End of stack trace from previous location where exception was thrown ---
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
            at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
            at Splunk.Client.AtomFeed.<ReadXmlAsync>d__40.MoveNext() in C:\Users\jdhea\Documents\Visual Studio 2015\Projects\splunk-sdk-csharp-pcl-2.2.2\src\Splunk.Client\Splunk\Client\AtomFeed.cs:line 259
         --- End of stack trace from previous location where exception was thrown ---
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
            at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
            at Splunk.Client.BaseEntity`1.<CreateAsync>d__18`1.MoveNext() in C:\Users\jdhea\Documents\Visual Studio 2015\Projects\splunk-sdk-csharp-pcl-2.2.2\src\Splunk.Client\Splunk\Client\BaseEntity.cs:line 221
         --- End of stack trace from previous location where exception was thrown ---
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
            at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
            at Splunk.Client.EntityCollection`2.<GetOrNullAsync>d__13.MoveNext() in C:\Users\jdhea\Documents\Visual Studio 2015\Projects\splunk-sdk-csharp-pcl-2.2.2\src\Splunk.Client\Splunk\Client\EntityCollection.cs:line 269
         --- End of stack trace from previous location where exception was thrown ---
            at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
            at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
            at Splunk.Examples.saved_searches.Program.<Run>d__1.MoveNext() in C:\Users\jdhea\Documents\Visual Studio 2015\Projects\splunk-sdk-csharp-pcl-2.2.2\examples\saved-searches\Program.cs:line 48
       InnerException:
```

**Comment by glennblock** [comment]
Thanks for reporting @machdevnet. We will look into it.

**Comment by quirket** [comment]
I am having the same issue. is there anymore information we can provide? or is there a workaround?

**Comment by glennblock** [comment]
@quirket have you tried our latest SDK? We fixed this specific issue I believe in 2.2.3. https://github.com/splunk/splunk-sdk-csharp-pcl/blob/master/CHANGELOG.md
